src/fastqsplitter/__init__.py

```diff
@@ -0,0 +1,137 @@
+                             "which compression algorithm will be used. '.gz' "
+                             "for gzip, '.bz2' for bzip2, '.xz' for xz. Other "
+                             "extensions will use no compression."
+                        )
+    parser.add_argument("-c", "--compression-level", type=int,
+                        default=DEFAULT_COMPRESSION_LEVEL,
+                        help="Only applicable when output files have a '.gz' "
+                             "extension. Default={0}"
+                        .format(DEFAULT_COMPRESSION_LEVEL))
+    parser.add_argument("-t", "--threads-per-file", type=int,
+                        default=DEFAULT_THREADS_PER_FILE,
+                        help="Set the number of compression threads per output"
+                             " file. NOTE: more threads are only useful when "
+                             "using a compression level > 1. Default={0}"
+                             "".format(DEFAULT_THREADS_PER_FILE))
+
+    # BELOW ARGUMENTS ARE FOR BENCHMARKING AND TESTING PURPOSES
+    parser.add_argument("-g", "--group-size", type=int,
+                        default=DEFAULT_GROUP_SIZE,
+                        help=argparse.SUPPRESS
+                        # help="Specify the group size. This will set how "
+                        #      "fine-grained the fastq distribution will be."
+                        #      " Default: {0}".format(DEFAULT_GROUP_SIZE)
+                        )
+
+    return parser
+
+
+def split_fastqs(input_file: Path, output_files: List[Path],
+                 compression_level: int = DEFAULT_COMPRESSION_LEVEL,
+                 group_size: int = DEFAULT_GROUP_SIZE,
+                 threads_per_file: int = DEFAULT_THREADS_PER_FILE):
+    # contextlib.Exitstack allows us to open multiple files at once which
+    # are automatically closed on error.
+    # https://stackoverflow.com/questions/19412376/open-a-list-of-files-using-with-as-context-manager
+    with contextlib.ExitStack() as stack:
+        input_fastq = stack.enter_context(
+            xopen.xopen(input_file, mode='rb'))  # type: xopen.PipedGzipReader
+        output_handles = [
+            stack.enter_context(xopen.xopen(
+                filename=output_file,
+                mode='wb',
+                compresslevel=compression_level,
+                threads=threads_per_file
+            )) for output_file in output_files
+        ]  # type: List[xopen.PipedGzipWriter]
+
+        i = 0
+        number_of_output_files = len(output_handles)
+        fastq_eof_reached = False
+        while not fastq_eof_reached:
+            # By using modulo we pick one file at a time and make sure all
+            # files are written to equally.
+            file_to_write = output_handles[i % number_of_output_files]
+
+            # Make sure a multiple of 4 lines is written as each record is
+            # 4 lines.
+            for j in range(group_size * 4):
```

Comment:
This necessarily does _not_ support fastq files with multiline sequences (those exist). That's fine - I wouldn't support this either - but it should be documented.

Reply:
It is mentioned in the README now.